kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/validation/JwsSignatureRecordValidationIT.java

```diff
@@ -0,0 +1,229 @@
+        ConfigurationBuilder config = createFilterDef(cluster, topic, false, false, false);
+
+        try (var tester = kroxyliciousTester(config);
+                var producer = tester.producer()) {
+            var result = producer.send(new ProducerRecord<>(topic.name(), null, null,
+                    EMPTY_STRING, RANDOM_STRING, List.of(new RecordHeader(JWS_HEADER_NAME, INVALID_JWS))));
+            assertThatFutureFails(result, InvalidRecordException.class,
+                    DEFAULT_ERROR_MESSAGE + ": Jose4j threw an exception: A JWS Compact Serialization must have exactly 3 parts");
+        }
+    }
+
+    @Test
+    void jwsSignedWithMissingJwkProduceRejected(KafkaCluster cluster, Topic topic) {
+        ConfigurationBuilder config = createFilterDef(cluster, topic, false, false, true);
+
+        try (var tester = kroxyliciousTester(config);
+                var producer = tester.producer()) {
+            var result = producer.send(new ProducerRecord<>(topic.name(), null, null,
+                    EMPTY_STRING, RANDOM_STRING, List.of(new RecordHeader(JWS_HEADER_NAME, VALID_JWS_USING_MISSING_ECDSA_JWK))));
+            assertThatFutureFails(result, InvalidRecordException.class,
+                    DEFAULT_ERROR_MESSAGE + ": Jose4j threw an exception: Could not select a valid JWK that matches the algorithm constraints");
+        }
+    }
+
+    @Test
+    void jwsSignedWithRsaJwkProduceRejectedDueToEcdsaConstraints(KafkaCluster cluster, Topic topic) {
+        ConfigurationBuilder config = createFilterDef(cluster, topic, false, false, true);
+
+        try (var tester = kroxyliciousTester(config);
+                var producer = tester.producer()) {
+            var result = producer.send(new ProducerRecord<>(topic.name(), null, null,
+                    EMPTY_STRING, VALID_JWS_USING_RSA_JWK_PAYLOAD, List.of(new RecordHeader(JWS_HEADER_NAME, VALID_JWS_USING_RSA_JWK))));
+            assertThatFutureFails(result, InvalidRecordException.class,
+                    DEFAULT_ERROR_MESSAGE + ": Jose4j threw an exception: Could not select a valid JWK that matches the algorithm constraints");
+        }
+    }
+
+    @Test
+    void invalidDetachedContentJwsProduceRejected(KafkaCluster cluster, Topic topic) {
+        ConfigurationBuilder config = createFilterDef(cluster, topic, false, true, true);
+
+        try (var tester = kroxyliciousTester(config);
+                var producer = tester.producer()) {
+            var result = producer.send(new ProducerRecord<>(topic.name(), null, null,
+                    EMPTY_STRING, RANDOM_STRING, List.of(new RecordHeader(JWS_HEADER_NAME, VALID_JWS_USING_ECDSA_JWK_AND_UNENCODED_CONTENT_DETACHED))));
+            assertThatFutureFails(result, InvalidRecordException.class, DEFAULT_ERROR_MESSAGE + ": JWS Signature was invalid");
+        }
+    }
+
+    private static void isJwsHeaderMatching(ConsumerRecord<String, String> record, byte[] value) {
```

Comment:
```suggestion
    private static void hasJwsHeaderWithValue(ConsumerRecord<String, String> record, byte[] value) {
```

`is` generally implies a boolean getter not a test. Where as this method is selecting a particular header and asserting that it has a specific value.

I'd have written it slightly differently as

```
    private static void isJwsHeaderMatching(ConsumerRecord<String, String> record, byte[] value) {
        assertThat(record.headers()).anySatisfy(header -> {
            assertThat(header.key()).isEqualTo(JWS_HEADER_NAME);
            assertThat(header.value()).isEqualTo(value);
        });
    }
    ```